src/sandstorm/supervisor.c++

```diff
@@ -1526,8 +1531,11 @@ void SupervisorMain::DefaultSystemConnector::checkIfAlreadyRunning() const {
   KJ_UNREACHABLE;
 }
 
-class SupervisorMain::SandstormApiImpl final: public SandstormApi<>::Server {
+class SupervisorMain::SandstormApiImpl final:
+  public SandstormApi<>::Server, public kj::Refcounted, private kj::TaskSet::ErrorHandler  {
```

Comment:
It's dangerous to make capability servers refcounted because then `thisCap()` (which is like `this` except returning a capability) doesn't work (due to implementation complications).

I think what you really want here is a separate class called `WakeLockSet` which is itself refcounted with `SandstormApiImpl` holding one reference. `WakeLockSet` would implement `restoreWakeLockNotification()` and such.